src/unix_dgram.cc

```diff
@@ -334,20 +334,29 @@ NAN_METHOD(Connect) {
   sockaddr_un s;
   int err;
   int fd;
+  socklen_t namelen;
+  unsigned int len;
 
   assert(info.Length() == 2);
 
   fd = info[0]->Int32Value();
   String::Utf8Value path(info[1]);
+  len = path.length();
+  if (len > sizeof(s.sun_path)) {
+    err = -EINVAL;
+    goto out;
+  }
 
   memset(&s, 0, sizeof(s));
-  strncpy(s.sun_path, *path, sizeof(s.sun_path) - 1);
+  memcpy(s.sun_path, *path, len);
```

Comment:
> Since abstract namespace sockets don't need to be null terminated the cases probably need to be handled separately 

It looks like it.

> The send method doesn't use a path did you mean the sendto method instead?

Yes